apps/swirl-docs/src/components/Navigation/SidebarNavigation/SidebarNavSubItem.tsx

```diff
@@ -0,0 +1,131 @@
+import icon from "@getflip/swirl-icons/icons/ChevronRight28.svg";
+import { NavItem } from "@swirl/lib/navigation/";
+import classNames from "classnames";
+import { AnimatePresence, motion } from "framer-motion";
+import Image from "next/image";
+import Link from "next/link";
+import { useEffect, useState } from "react";
+import { ANIMATION_CLOSED, ANIMATION_OPENED } from ".";
+import { SidebarNavItem } from "./SidebarNavItem";
+
+export const SidebarNavSubItem = ({
```

Comment:
I think the structure makes it hard to read and understand which navItem is rendered when. 

The ternaries for the conditional rendering are just too deeply nested. 

A few weeks ago i tidied up the mobile navigation to use a strategy pattern for recursevly rendering the navigation and i think this strategy can be also applied here. They were in the `HeaderNavigation/Mobile` package. 

There are several benefits for doing that:
- Makes it easier to understand what the current component is (they are all sidebarnavsubitems atm)
- makes the code more maintainable and extendable in the future

Reply:
restructured navigation, please have a look